lambdas/services/process_mns_message_service.py

```diff
@@ -0,0 +1,140 @@
+import os
+
+from botocore.exceptions import ClientError
+from enums.death_notification_status import DeathNotificationStatus
+from enums.mns_notification_types import MNSNotificationTypes
+from enums.patient_ods_inactive_status import PatientOdsInactiveStatus
+from models.mns_sqs_message import MNSSQSMessage
+from services.base.dynamo_service import DynamoDBService
+from services.base.sqs_service import SQSService
+from utils.audit_logging_setup import LoggingService
+from utils.exceptions import PdsErrorException
+from utils.utilities import get_pds_service
+
+logger = LoggingService(__name__)
+
+
+class MNSNotificationService:
+    def __init__(self):
+        self.message = None
+        self.dynamo_service = DynamoDBService()
+        self.table = os.getenv("LLOYD_GEORGE_DYNAMODB_NAME")
+        self.pds_service = get_pds_service()
+        self.sqs_service = SQSService()
+        self.queue = os.getenv("MNS_NOTIFICATION_QUEUE_URL")
+
+    def handle_mns_notification(self, message: MNSSQSMessage):
+        try:
+            if message.type == MNSNotificationTypes.CHANGE_OF_GP.value:
+                logger.info("Handling GP change notification.")
+                self.handle_gp_change_notification(message)
+
+            elif message.type == MNSNotificationTypes.DEATH_NOTIFICATION.value:
+                logger.info("Handling death status notification.")
+                self.handle_death_notification(message)
+
+        except PdsErrorException:
+            self.send_message_back_to_queue(message)
+
+        except Exception as e:
```

Comment:
I think we should catching a general Exception only in one place. 
So can you change this to a more specific one?